### ISSUE TITLE
Assertion failed: CurrentIndex+Count<=CompressedData.Num()

### ISSUE THREAD

**Issue by CptnCanuck** [wolveix/satisfactory-server#376]
### Describe the Bug

Server seems to be crashing on initialization, and can't recover to get up and running. I'm unable to parse the logs myself and understand where things my be going wrong, but the server was working fine up until yesterday. 
On December 1st I logged in and played the first day of the Ficsmas event without any issues, but upon trying to log in yesterday (Dec 2nd), the docker container was running into a number of issues. I was first receiving numerous permissions errors relating to an inability to write to the config directory, which was odd because that container had been running fine just the day before. Still, I updated the config folder owner to match the ID I'm running the docker container with (1000), and those errors seemed to dissipate. 
Still, when I run the container, it attempts to initialize, updates the game, (seems to be confirmed by log files containing: _(Build: ++FactoryGame+rel-main-ficsmas-2024-CL-382498)_) and then appears to hit a crash, which it logs, and then restarts. This repeats until I kill the container. 

### Your Runtime Command or Docker Compose File

```
services:
  satisfactory-server:
    container_name: 'satisfactory-server'
    hostname: 'satisfactory-server'
    image: 'wolveix/satisfactory-server:latest'
    ports:
      - '7777:7777/udp'
      - '7777:7777/tcp'
    volumes:
      - 'volumemappinghere'
    environment:
      - MAXPLAYERS=4
      - PGID=1000
      - PUID=1000
      - ROOTLESS=false
      - STEAMBETA=false
    restart: unless-stopped
    deploy:
      resources:
        limits:
          memory: 6G
        reservations:
          memory: 4G
```

### Debug Output

[debug_dump.txt](https://github.com/user-attachments/files/17989795/debug_dump.txt)

### System Specs (please complete the following information):

```shell
===== START ISSUE REPORT =====
OS:  Linux PlexBox 6.1.0-28-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.119-1 (2024-11-22) x86_64 GNU/Linux
CPU: Intel(R) Core(TM) i7-4790K CPU @ 4.00GHz
RAM: 5GB/15GB
HDD: 63GB/227GB (29% used)
===== END ISSUE REPORT =====
```

### Logs

[container_log.txt](https://github.com/user-attachments/files/17989862/container_log.txt)

### Additional Context

Entirely possible this is user error or something I didn't know I did. Relatively new to working with docker and linux in general. Have figured out enough to get this old gaming machine running my plex server as well as this satisfactory server, but I'm sure the setup is not ideal, and I may have absolutely misconfigured something, or failed to realize that I broke something while tinkering. Please don't hesitate to ask the obvious questions, though I have turned the machine off and on again. 😅

**Comment by wolveix** [comment]
Hey @CptnCanuck! It seems your log is missing the first few lines, so I can't see what version you're running, though I expect it's 1.8.9. Additionally, it would be useful to see your full mapping :) It's very likely a game bug/related to the permissions issues you experienced, so please try the following:

I'd stop the container completely, remove it (`docker rm satisfactory-server`), then delete the `/config/gamefiles` directory completely. Then re-launch the container, and see how it goes. If the same issue presents, I'd stop and remove the container again, backup/move your existing host-bound `/config` directory, and start the container up completely fresh. There's a possibility your save experienced an issue, which shouldn't be a big issue as it auto-backs up (and could be repaired through the Satisfactory Calculator site) :) 

Let me know how you get on!

**Comment by CptnCanuck** [comment]
Hey! Thanks for the support. Looks like it was a corrupted save. Rebuilt the entire container from scratch and when I re-uploaded my save file, it reverted to the crashing behaviour. Looked like I just needed to roll back to a save from a couple hours earlier that night. Once I did I was up and running with no issues. 

Must be related to the Ficsmas event somehow. I'll try to soldier on and hopefully don't encounter it again. 

Thanks so much for the rapid response and helpful insights. You're a beauty.
